core/hydromet_stratified.py

```diff
@@ -289,13 +286,14 @@ def weights_Rainfall(Return_Intervals: np.ndarray, GEV_parameters: np.ndarray, P
     weights_R_topdown = (1.0/Bin_Bounds_R_topdown[:-1]-1.0/Bin_Bounds_R_topdown[1:]).astype(float)
     weights_R_topdown = np.append(weights_R_topdown, 1.0/RI_upper_bound)
     data = np.vstack((Bin_Bounds_R_topdown, np.append(Bin_Bounds_R_topdown[1:], np.inf), weights_R_topdown)).T
-    df_weights = pd.DataFrame(data=data, index=RI_index, columns=['Bin Floor', 'Bin Ceiling', 'Event Weight']) 
+    df_weights = pd.DataFrame(data=data, index=RI_index, columns=['Bin Floor', 'Bin Ceiling', 'Event Weight'])
+    RI_data =NOAA_precip[NOAA_precip.index.isin(Return_Intervals)].index.to_numpy().astype(int)
     RI_index_calc = RI_index[np.isin(RI_index, RI_data, invert=True)]
     Precip_calculate = GEV_RI(RI_index_calc, GEV_parameters, PMP)
     df2 = pd.DataFrame(data = Precip_calculate, index = RI_index_calc, columns=[ID])  
     df_R_NOAA_E = NOAA_precip[NOAA_precip.index.isin(RI_data)].copy()
     df_precip = df_R_NOAA_E.append(df2)
-    df_precip = df_precip.drop('P_Median_in', axis=1)
+    df_precip = pd.DataFrame(df_precip.iloc[:,0])#df_precip.drop('P_Median_in', axis=1)
```

Comment:
Better to avoid hard coding and pass the variable name when possible. Here instead of selecting the zeroth element within the dataframe, I would do pd.DataFrame(df_precip[ID])

Reply:
Also, be careful about dividing by integers. It is always safer to divide by a float, e.g. use 3.0 instead of 3.

Reply:
Note, resolved. Thanks!